indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractFixedIntervalTask.java

```diff
@@ -77,10 +77,15 @@ protected AbstractFixedIntervalTask(
 
   @Override
   public boolean isReady(TaskActionClient taskActionClient) throws Exception
+  {
+    return isReady(taskActionClient, TaskLockType.EXCLUSIVE);
+  }
+
+  protected boolean isReady(TaskActionClient taskActionClient, TaskLockType taskLockType) throws Exception
```

Comment:
Can we leave the base implementation as it is, without adding this extra protected method? The logic is simple enough for child class implementations to fully override `isReady()` without calling `super.isReady()`

Reply:
@abhishekrb19 thanks for the suggestion, wdyt about following:
parent's isReady() uses private field of `interval`, so either we convert it to protected or leave some base method
do you think it will be better?

Reply:
@IgorBerman, I see, you can just call the public getter `getInterval()` from the base class which is already used by the kill task

Reply:
updated